docs/proposals/local-cluster-ux.md

```diff
@@ -0,0 +1,184 @@
+Since we are targeting a single command install/teardown experience, vagrant needs to be an implementation detail and not be exposed to our users.
+
+## Proposed Solution
+
+To avoid exposing users to third party software and external dependencies, we will build a toolbox that will be shipped with all the dependencies including all kubernetes components, hypervisor, base image, kubectl, etc.
+*Note: Docker provides a [similar toolbox](https://www.docker.com/products/docker-toolbox).*
+This "Localkube" tool will be referred to as "Nanokube" in this proposal to avoid ambiguity against Spread's existing ["localkube"](https://github.com/redspread/localkube).
+The final name of this tool is TBD. Suggestions are welcome!
+
+Nanokube will provide a unified CLI to interact with the local cluster.
+The CLI will support only a few operations:
+    - **Start** - creates & starts a local cluster along with setting up kubectl & networking (if necessary)
+    - **Stop** - suspends the local cluster & preserves cluster state
+    - **Delete** - deletes the local cluster completely
+    - **Upgrade** - upgrades internal components to the latest available version (upgrades are not guaranteed to preserve cluster state)
+
+For running and managing the kubernetes components themselves,  we can re-use [Spread's localkube](https://github.com/redspread/localkube).
+Localkube is a self-contained go binary that includes all the master components including DNS and runs them using multiple go threads.
+Each Kubernetes release will include a localkube binary that has been tested exhaustively.
+
+To support Windows and OS X, nanokube will use [libmachine](https://github.com/docker/machine/tree/master/libmachine) internally to create and destroy virtual machines.
+Nanokube will be shipped with an hypervisor (virtualbox) in the case of OS X.
+Nanokube will include a base image that will be well tested.
+
+In the case of Linux, since the cluster can be run locally, we ideally want to avoid setting up a VM.
+Since docker is the only fully supported runtime as of Kubernetes v1.2, we can initially use docker to run and manage localkube.
+There is risk of being incompatible with the existing version of docker.
+By using a VM, we can avoid such incompatibility issues though.
+Feedback from the community will be helpful here.
+
+If the goal is to run outside of a VM, we can have nanokube prompt the user if docker is unavailable or version is incompatible.
+Alternatives to docker for running the localkube core includes using [rkt](https://coreos.com/rkt/docs/latest/), setting up systemd services, or a System V Init script depending on the distro.
+
+To summarize the pipeline is as follows:
+
+##### OS X / Windows
+
+nanokube -> libmachine -> virtualbox/hyper V -> linux VM -> localkube
+
+##### Linux
+
+nanokube -> docker -> localkube
```

Comment:
@runseb Mostly yes. The intention is to use `libmachine` and not `docker-machine`. This implementation can change in the future.